content/engineering-practices-rotting-pull-requests.md

```diff
@@ -0,0 +1,72 @@
+
+> ![Visualizing a growing backlog in the form of an overloaded task board](/images/engineering-practices-rotting-pull-requests-1.png)
+
+The exasperation of too many pull requests in-flight clogs the workflow. The chaotic backlog and continuous arrival of new features highlight the urgency and delays in merging changes to improve system outcomes.
+
+
+It’s not a good place to be. We need to get our processes back on track to deliver what the business needs and expects—at a high level of verified quality and functionality, without introducing vulnerabilities.
+
+---
+
+## The Hot Seat: Addressing 900+ Rotting Pull Requests
+
+My manager came to me today, asking: **How can we promote good practices and get the 900+ rotting pull requests actioned to reduce the backlog?**
+
+This is the subject of today’s post—and the hot seat we share.
+
+### Where Did All These Pull Requests Come From?
+
+The first questions should be: 
+- Where did all these pull requests come from? 
+- Why are they not being actioned? 
+
+Are teams too busy adding features to perform necessary code reviews and foster team collaboration?
+
+### Quality and the Feature Farming Trap
+
+What happens to quality when teams act like *feature farmers*, focusing only on the next big functionality without seeing the big picture? Yes, Houston, we have a problem.
+
+We have a massive codebase—over 300 applications and growing. Many of these applications are aging and of questionable utility, requiring upgrades to avoid obsolescence and unsupported components.
+
+To address this, we’ve created automation that generates the needed upgrade changes and creates pull requests for engineers to review before committing changes to the repository. It’s an odd place to be, having code enhanced automatically and then reviewed by a human to ensure quality.
+
+---
+
+## Automation and the Paradigm Shift
+
+Typically, we write tests to verify that changes meet functional (and non-functional) requirements. However, this is different. We’re now reviewing and approving code changes generated by a program for another program—it just feels weird. It’s a mind-shift.
+
+In the age of Artificial Intelligence (AI), our existing paradigms are shifting and evolving. We must ensure our processes are solid yet malleable, with even higher levels of quality and functionality. To achieve this, we need to step back from the *feature factory* we’ve become and reevaluate how we operate.
+> ![Engineers moving from old paradigm to a Software Factory using AI to get to the next level](/images/engineering-practices-rotting-pull-requests-1.png)
+
+---
+
+## The Path Forward: Embracing Automation and Addressing Backlogs
+
+This paradigm shift requires us to start viewing our automation as a **technical advantage**—an enabler to grow beyond our current struggles. But first, we **must** address our backlog of technical debt. Awareness is the first step before meaningful change can occur.
```

Comment:
@MerlinDarkstar "technical" debt will poke some hornets nest.

Can we say <*> debt, where * is one of more of technical, technology, code, test, or other, based on your definition of tech debt.